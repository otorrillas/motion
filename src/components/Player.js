import React from 'react';

import styles from './Player.module.css';

const Player = React.forwardRef((props, ref) => {
  const { track } = props;
  const src = track ? `tracks/${track.id}.mp3` : null;

  return (
    <audio src={src} controls ref={ref} autoPlay className={styles.player} />
  );
});

export default Player;
