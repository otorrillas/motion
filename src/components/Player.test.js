import React from 'react';
import renderer from 'react-test-renderer';

import Player from './Player';

describe('<Player />', () => {
  const requiredProps = {
    track: {
      id: 123
    }
  };
  const renderComponent = props =>
    renderer.create(<Player {...requiredProps} {...props} />);

  describe('@render', () => {
    it('default', () => {
      const tree = renderComponent();
      expect(tree).toMatchSnapshot();
    });
  });
});
