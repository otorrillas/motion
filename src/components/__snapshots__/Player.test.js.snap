// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> @render default 1`] = `
<audio
  autoPlay={true}
  className="player"
  controls={true}
  src="tracks/123.mp3"
/>
`;
